modules/net-ilb/main.tf

```diff
@@ -80,11 +80,7 @@ resource "google_compute_region_backend_service" "default" {
         : null
       )
       idle_timeout_sec = local.bs_conntrack.idle_timeout_sec
-      tracking_mode = (
-        local.bs_conntrack.track_per_session != null
-        ? local.bs_conntrack.track_per_session
-        : null
-      )
+      tracking_mode    = try(local.bs_conntrack.track_per_session ? "PER_SESSION" : "PER_CONNECTION", null)
```

Comment:
this should probably be

`try(local.bs_conntrack.track_per_session, null) ? "PER_SESSION" : "PER_CONNECTION"`

Reply:
The suggested code will work only if track_per_session is either true or false, in case it is null it returns error because the condition value will be null ("The condition value is null. Conditions must either be true or false."). The try statement implemented will check if the variable is true or false and assign the value accordingly, in case it is null the first expression will result in an error (same of the previous case) and the try will assign value null to the tracking_mode (null will then result in the default value of "PER_CONNECTION". Am I missing something?

Reply:
nope, you're right and I should not multitask :)